client/client_test.go

```diff
@@ -193,32 +195,52 @@ func TestAddLeaf(t *testing.T) {
 	cli := trillian.NewTrillianLogClient(env.ClientConn)
 	for _, test := range []struct {
 		desc    string
+		leaf    []byte
 		client  trillian.TrillianLogClient
 		wantErr bool
 	}{
-		{
-			desc:   "success 1",
-			client: cli,
-		},
-		{
-			desc:   "success 2",
-			client: cli,
-		},
-		{
-			desc:    "invalid inclusion proof",
-			client:  &MockLogClient{c: cli, mGetInclusionProof: true},
-			wantErr: true,
-		},
-		{
-			desc:    "invalid consistency proof",
-			client:  &MockLogClient{c: cli, mGetConsistencyProof: true},
-			wantErr: true,
-		},
+		{desc: "First leaf", leaf: []byte("A"), client: cli},
+		{desc: "Make TreeSize > 1", leaf: []byte("B"), client: cli},
 	} {
-		client := New(logID, test.client, testonly.Hasher, env.PublicKey)
-		err := client.AddLeaf(ctx, []byte(test.desc))
+		client := New(logID, test.client, rfc6962.DefaultHasher, env.PublicKey)
+		if err := client.QueueLeaf(ctx, test.leaf); err != nil {
+			t.Errorf("QueueLeaf(%v): %v", test.desc, err)
+		}
+		env.Sequencer.OperationSingle(ctx)
+	}
+}
+
+func TestWaitForInclusion(t *testing.T) {
+	ctx := context.Background()
+	env, err := integration.NewLogEnv(ctx, 0, "TestWaitForInclusion")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer env.Close()
+	logID, err := env.CreateLog()
+	if err != nil {
+		t.Fatalf("Failed to create log: %v", err)
+	}
+
+	cli := trillian.NewTrillianLogClient(env.ClientConn)
+	for _, test := range []struct {
+		desc    string
+		leaf    []byte
+		client  trillian.TrillianLogClient
+		wantErr bool
+	}{
+		{desc: "First leaf", leaf: []byte("A"), client: cli},
+		{desc: "Make TreeSize > 1", leaf: []byte("B"), client: cli},
+		{desc: "invalid inclusion proof", leaf: []byte("A"), client: &MockLogClient{c: cli, mGetInclusionProof: true}, wantErr: true},
+	} {
```

Comment:
(At the moment these tests completely subsume the ones in `TestQueueLeaf` so the latter could be dropped.  However, if you add some error cases above that will no longer be the case and you can ignore this comment.)